pkg/app/pipedv1/plugin/kubernetes/provider/loader.go

```diff
@@ -85,13 +85,23 @@ func NewLoader(registry ToolRegistry) *Loader {
 
 func (l *Loader) LoadManifests(ctx context.Context, input LoaderInput) (manifests []Manifest, err error) {
 	defer func() {
-		// Add builtin annotations for tracking application live state.
 		for i := range manifests {
+			// Set the namespace for all manifests.
+			manifests[i].body.SetNamespace(input.Namespace)
```

Comment:
We need this to ensure the loaded manifests' namespaces are matched with applied resource.

Reply:
Oops, we have to check the `input.Namespace` is empty or not.